crates/sui/src/client_commands.rs

```diff
@@ -1115,22 +1125,31 @@ impl SuiClientCommands {
                     if modules.len() != 1 { "s" } else { "" },
                 );
 
-                // Need to pass some bounds because otherwise this meter doesn't accumulate
-                // anything.
-                let mut meter = BoundMeter::new(MeterConfig {
-                    max_per_fun_meter_units: Some(u128::MAX),
-                    max_per_mod_meter_units: Some(u128::MAX),
-                    max_per_pkg_meter_units: Some(u128::MAX),
-                });
-
+                let mut meter = AccumulatingMeter::new();
                 verifier.meter_compiled_modules(&protocol_config, &modules, &mut meter)?;
 
+                let mut used_ticks = meter.accumulator(Scope::Package).clone();
+                used_ticks.name = pkg_name;
+
                 let meter_config = protocol_config.meter_config();
+
+                let exceeded = matches!(
+                    meter_config.max_per_pkg_meter_units,
+                    Some(ticks) if ticks < used_ticks.max_ticks(Scope::Package)
```

Comment:
nit just for quick reading maybe
```suggestion
                    Some(allowed_ticks) if allowed_ticks < used_ticks.max_ticks(Scope::Package)
```